SofaKernel/modules/SofaBaseMechanics/SofaBaseMechanics_test/DiagonalMass_test.cpp

```diff
@@ -141,7 +146,9 @@ class DiagonalMass_test : public ::testing::Test
         string scene =
                 "<?xml version='1.0'?>"
                 "<Node 	name='Root' gravity='0 0 0' time='0' animate='0'   > "
-                "   <MechanicalObject position='0 0 0 4 5 6'/>               "
+                "    <MechanicalObject />                                                                       "
+                "    <RegularGrid nx='2' ny='2' nz='2' xmin='0' xmax='2' ymin='0' ymax='2' zmin='0' zmax='2' /> "
+                "    <HexahedronSetGeometryAlgorithms />                                                        "
                 "   <DiagonalMass name='m_mass'/>                            "
```

Comment:
The topology is needed because otherwise the component will emit a msg_error(). As the PR make the test in-herit from Sofa_test the error message would be converted to test failure. 

I didn't wanted to add the line: 
EXPECT_MSG_EMIT(Error) and set the object into an invalid state. 
For this specific test I think it was more correct to add the topology so the object can be inited correctly and be tested.